op-service/client/ethclient.go

```diff
@@ -16,3 +17,17 @@ func DialEthClientWithTimeout(ctx context.Context, url string, timeout time.Dura
 
 	return ethclient.DialContext(ctxt, url)
 }
+
+// DialEthClientWithRetry attempts to dial the L1 provider using the provided
+// URL. If the dial fails, this method will retry the dial every interval seconds
+// until the context is timed out.
+func DialEthClientWithRetry(ctx context.Context, url string, strategy backoff.Strategy, maxAttempts int, timeout time.Duration) (*ethclient.Client, error) {
+	operation := func() (*ethclient.Client, error) {
+		ctxt, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+
+		return ethclient.DialContext(ctxt, url)
+	}
+
+	return backoff.DoResult(maxAttempts, strategy, operation)
```

Comment:
With a `http` connection this won't actually do anything because the `DialContext` method doesn't attempt to make a connection.  They're just made on demand when a request is made later.